webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java

```diff
@@ -138,7 +138,13 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
                 send100Continue(ctx);
             }
 
-            routing.route(bareRequest, bareResponse);
+            // If a problem during routing, return 400 response
+            try {
+                routing.route(bareRequest, bareResponse);
+            } catch (IllegalArgumentException e) {
```

Comment:
Just wondering... Are there other exceptions besides `IllegalArgumentException` that might be thrown that should also trigger the bad request response? I see that the checking at line 111 is the same.

Reply:
I haven't encounter other exceptions, but perhaps. This is intended to be an improvement over what we have now, but perhaps not the last one of these.